de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/impexp/BPartnerImportProcess.java

```diff
@@ -166,10 +161,12 @@ else if (importRecord.getC_BPartner_ID() <= 0 || importRecord.getC_BPartner_ID()
 			}
 		}
 
-		createOrUpdateBPartnerLocation(importRecord, sameBPpreviousImportRecords);
+		createOrUpdateBPartnerLocation(importRecord, context.getSameBPpreviousImportRecords());
 		createUpdateContact(importRecord);
 		createUpdateInterestArea(importRecord);
 
+		context.getSameBPpreviousImportRecords().add(importRecord); // set in the context the new line
```

Comment:
avoid this. According to law of Demeter, you shall have only one dot.
That translates to something like:
```
context.collectImportRecord(importRecord);
```